lib/internals.js

```diff
@@ -0,0 +1,930 @@
+      };
+    },
+
+    // `id` is `public` or `apos`
+    hasViteBuildManifest(id) {
+      return fs.existsSync(self.buildManifestPath[id]);
+    },
+
+    // Filter the entrypoints for different devServer scenarios in development.
+    // The build option `devServer` can be `public` or `apos`. We want to filter
+    // the entrypoints based on that.
+    // `id` is `public` or `apos`
+    // TODO: filtering `bundled` by type does not scale well. We need to introduce
+    // a `bundled: Boolean` property to the entrypoint configuration in the future.
+    // Also we might need a specific `buildTag` (or better name) that corresponds
+    // to `devServer` and `types` (can be string 'public' or 'apos'). This will allow us
+    // to introduce new entrypoint types and features without breaking the current logic.
+    getBuildEntrypointsFor(id) {
+      if (id === 'apos') {
+        return self.entrypointsManifest
+          .filter((entrypoint) => entrypoint.type === 'apos');
+      }
+      if (id === 'public') {
+        return self.entrypointsManifest
+          .filter((entrypoint) => ![ 'bundled', 'apos' ].includes(entrypoint.type));
+      }
+      throw new Error(`Invalid build ID "${id}"`);
+    },
+
+    // Return the configuration and the vite build function for a given build scenario.
+    // `id` is `public` or `apos`
+    async getViteBuild(id, options) {
+      const { build } = await import('vite');
+      const config = await self.getViteConfig(id, options);
+      return {
+        build,
+        config
+      };
+    },
+
+    // Get the Inline Vite configuration for a given build scenario.
+    // https://vite.dev/guide/api-javascript.html#inlineconfig
+    // This is the high level method that should be used to get the Vite configuration.
+    // `id` is `public` or `apos`.
+    // `options` is build options.
+    // `command` is `build` or `serve`.
+    async getViteConfig(id, options, command = 'build') {
+      const env = {
+        command,
+        mode: self.apos.asset.isProductionMode() ? 'production' : 'development'
+      };
+      const baseConfig = await self.getBaseViteConfig(id, options, env);
+
+      /** @type {import('vite').UserConfig} */
+      let resolved;
+      if (id === 'public') {
+        resolved = await (await self.getPublicViteConfig(baseConfig))(env);
+      }
+      if (id === 'apos') {
+        resolved = await (await self.getAposViteConfig(baseConfig))(env);
+      }
+
+      if (!resolved) {
+        throw new Error(`Invalid Vite config ID "${id}"`);
+      }
+
+      return self.getFinalViteConfig(id, options, resolved, env);
+    },
+
+    // Return the input configuration for the Vite build for a given build scenario.
+    // `id` is `public` or `apos`.
+    getBuildInputs(id) {
+      return Object.fromEntries(
+        self.getBuildEntrypointsFor(id)
+          .map((entrypoint) => ([
+            entrypoint.name,
+            path.join(self.buildRootSource, `${entrypoint.name}.js`)
```

Comment:
can we guarantee that the entrypoint name will not already contain `.js` at the end, or be something else?